packages/build/src/plugins_core/functions/error.js

```diff
@@ -17,30 +23,65 @@ const getZipError = function (error, functionsSrc) {
   return error
 }
 
-const isModuleNotFoundError = function (error) {
-  return error instanceof Error && error.code === MODULE_NOT_FOUND_CODE
+const getModuleNotFoundError = async function (error, functionsSrc) {
+  const errorFromZisi = await getModuleNotFoundErrorFromZISI(error, functionsSrc)
+
+  if (errorFromZisi) {
+    return errorFromZisi
+  }
+
+  const errorFromEsbuild = await getModuleNotFoundErrorFromEsbuild(error, functionsSrc)
+
+  if (errorFromEsbuild) {
+    return errorFromEsbuild
+  }
 }
 
-const MODULE_NOT_FOUND_CODE = 'MODULE_NOT_FOUND'
+const getModuleNotFoundErrorFromEsbuild = async function (error, functionsSrc) {
+  const { errors = [] } = error
+  const modulesNotFound = errors.reduce((modules, errorObject) => {
+    const match = errorObject.text.match(MODULE_NOT_FOUND_ESBUILD_REGEXP)
+
+    if (!match) {
+      return modules
+    }
+
+    return [...modules, match[1]]
+  }, [])
+
+  if (modulesNotFound.length === 0) {
+    return
+  }
+
+  const isMissingNodeModulesDirectory = await lacksNodeModules(functionsSrc)
+  const message = isMissingNodeModulesDirectory ? getLocalInstallMessage(modulesNotFound) : MODULE_NOT_FOUND_MESSAGE
 
-const getModuleNotFoundError = async function (error, functionsSrc) {
-  const message = await getModuleNotFoundMessage(error, functionsSrc)
   error.message = `${message}\n\n${error.message}`
   addErrorInfo(error, { type: 'dependencies' })
```

Comment:
[sand] Would there be an opportunity to share the common code between `getModuleNotFoundErrorFromZISI()` and `getModuleNotFoundErrorFromEsbuild()`?

Reply:
Sure! Done in https://github.com/netlify/build/pull/2361/commits/e46268cc26345398f6827799a555310f7277db1e.